File: php/features.php
<?php
declare(strict_types=1);

// MeteoprogWeather SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MeteoprogWeatherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MeteoprogWeatherBaseFeature();
            case "test":
                return new MeteoprogWeatherTestFeature();
            default:
                return new MeteoprogWeatherBaseFeature();
        }
    }
}

File: py/meteoprogweather_sdk/features.py
# MeteoprogWeather SDK feature factory

from meteoprogweather_sdk.feature.base_feature import MeteoprogWeatherBaseFeature
from meteoprogweather_sdk.feature.test_feature import MeteoprogWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MeteoprogWeatherBaseFeature(),
        "test": lambda: MeteoprogWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
